### DISCUSSION TITLE
Custom Block Breaking Times

### DISCUSSION THREAD

**Discussion by Tofaa2** [Minestom/Minestom#1720] [Help] [answered]
Anyone know how i would go on about approaching custom breaking times for a specific block? For example, i want the diamond ore to be able to be mined within  a second in survival but also maintain the survival like mining animation and particles

**Comment by mworzala** [comment, marked as answer]
You give the player a high mining fatigue and then listen for them to start digging and manually send the block break animation packets at the interval you want. One common pitfall here is that you need to send the break packets for an entity id that is NOT the player (eg player entity id + 1 should work).

**Reply by Tofaa2** [reply]
Would simply cancelling the block breaking attempt work instead of mining fatigue? Then proceed to check if theyre still mining it or not

**Reply by mworzala** [reply]
You need the client to remain thinking it is mining so that you can listen for dig cancel to cancel the effect. If the client keeps mining then you will get fighting with the break particles (or the predicted ones will just override yours), and the client will predict the block breaking at the wrong time.

**Reply by Tofaa2** [reply]
Hmm fair, i think that issue is seen in certain servers such as hypixel when they attempt mining/altering mining in general. Thanks for the clearance

**Reply by Flopgop** [reply]
Would sending `Integer.MAX_VALUE` as the entity id work?

**Reply by mworzala** [reply]
Anything is valid for the player besides the player themselves. It gets slightly more tricky if you want to also make the break animation visible for another player. In that case you need different IDs for the different players.

**Reply by Tofaa2** [reply]
How would i go about tracking if the player is still breaking the block. Also is the breaking packet the EffectsPacket or is there some other packet i should be aware of
